File: 2020/code/day1.py
data_file = "../data/day1.txt"

# get data
data = open(data_file).readlines()
data = [int(d) for d in data]

# part 1 - product of the 2 values sum to 2020

for i in range(len(data)):
    n1 = data[i]
    for n2 in data[i:]:   
        nsum = n1 + n2
        if nsum == 2020:
            print(f"the numbers are {n1} and {n2}")
            print(f"their product is {n1*n2}")
            break

# part 2 - product of the 3 values sum to 2020

for i in range(len(data)):
    n1 = data[i]
    for j in range(len(data[i:])):  
        n2 = data[i+j] 
        for n3 in data[i+j:]:
            nsum = n1 + n2 + n3
            if nsum == 2020:
                print(f"the numbers are {n1}, {n2}, and {n3}")
                print(f"their product is {n1*n2*n3}")
                break

File: 2020/code/day1.R
library(data.table)

data_file <- "../data/day1.txt" 
data <- fread(data_file)

# part 1 - product of 2 numbers whose sum is 2020

combos2 <- data.table(t(combn(data[[1]], 2)))
combos2[,sum := V1 + V2]
ans = combos2[sum == 2020, .(V1, V2, prod = V1*V2)]

print(paste("the values are", ans[,V1], "and", ans[,V2], 
            "and the product is", ans[,prod]))

# part 2 - product of 3 numbers whose sum is 2020

combos3 <- data.table(t(combn(data[[1]], 3)))
combos3[,sum := V1 + V2 + V3]
ans = combos3[sum == 2020, .(V1, V2, V3, prod = V1*V2*V3)]

print(paste("the values are", ans[,V1], "and", ans[,V2], "and", ans[,V3], 
            "and the product is", ans[,prod]))
